frame/support/procedural/src/lib.rs

```diff
@@ -583,3 +583,23 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
 		.unwrap_or_else(|r| r.into_compile_error())
 		.into()
 }
+
+mod kw {
+	syn::custom_keyword!(cached);
+}
+
+#[proc_macro_attribute]
+pub fn benchmarking(attr: TokenStream, item: TokenStream) -> TokenStream {
```

Comment:
After reading the rest of the code. I don't get why we need this? Panicking is a bad idea as I already said. If we parse this in the `pallet` macro, you don't need this.

Reply:
Yes this is only here to ensure that the macro can only be attached to valid storage items. Happy to rework as I mentioned above if we are ok with calling it `#[pallet::whitelist]` or `#[pallet::storage::whitelist]` if that is desirable? This might also resolve the pub crate issue cc @shawntabrizi

Reply:
`#[pallet::storage_whitelist]` sounds good to me.

Reply:
I don't get why we need to call it `pallet::storage_whitelist`? We could still go with `benchmarking(cached)`. `storage_whitelist` sounds too generic to me. I will not die on this hill and don't care that much as long as you reorganize the code.

Reply:
I've gone ahead and restructured things so that we use a quoted `compile_error!` instead in both places and also moved the storage_alias specific code into `storage_alias.rs` and pulled it out as a `validate_storage_item` function, so hopefully now things look good without needing to rename the macro.